sdgx/synthesizer.py

```diff
@@ -389,15 +393,26 @@ def _sample_once(
 
         """
         missing_count = count
-        max_trails = 50
+        max_trails = 5
```

Comment:
Why it's change back to `5` and `missing_count * 1.2` ? We need more `max_trails`.

Reply:
For small dataset, *4 may cause long time to generate especially in CTGAN? I have used the batch size in once sampling number to improve 'max_trails'. for other model that may need more trail?


Can i use this following?
max_trails = 50
batched model -> *1.2 for once sample
non_batched model -> *4 for once sample

Reply:
> For small dataset, *4 may cause long time to generate especially in CTGAN? I have used the batch size in once sampling number to improve 'max_trails'. for other model that may need more trail?
> 
> Can i use this following? max_trails = 50 batched model -> *1.2 for once sample non_batched model -> *4 for once sample

**Background of the issue I hava**
Once we sampling from model, it's actually a post-process by `procrssors`, which like:
```python
            for d in self.data_processors:
                sample_data = d.reverse_convert(sample_data)
```

And I find that if we set max_trails to 5, the sampled data will be much smaller than the sample size specified by the user (In some cases of data).

**Question**
So if the `batch_size` is increased, will there still be enough sample size after processing and filtering by the processors? This is what I want to understand and confirm.

Reply:
For why it's 1.2 not 4.
If we need 1000 sample, using 1.2 and batchsize=500, it will raw generated 1500 (cause we use 1200 to sample firstly), it still have 500 remain to filter. In some time, it enough.
But if we use 4, in that situation, it will take 2.5x extra times.

Reply:
> Now let's talk the batchsize.
> 
> Let's think for the original design. When the missing count decreasing by cycle, only to multiply it is not a good plan, cause it converge to zero. Such as the missing=50, we use _4, model raw generated 200, then when reversed, it also has a probability still to missing. But in actually, the batched model can only generated by whole batch, if the size=500, 500-200=300 samples are ignored. If we select the max of missing_multiply and batchsize, it can reduce ignored sample, and reduce the missing probability after this trail.
> 
> So, i add 'drop more' argument in ctgan, then in sampling if the missing_multiply is not a multiple of batch size, it will be a ceil of it. And if missing_multiply < batchsize, it will be a batchsize.
> 
> So, for ctgan, we can also remove this code change. But for other model in batched, this method can do some favor, if the model not has drop_more (such as tvae, we can create an issue).

I see. To summarize, this issue arises from the lack of variable association between the `batch_size` of `Synthesizer.sample()` and `model.sample()`, resulting in the `batch_size` set by Synthesizer being smaller than the number of samples from `model.sample()`, causing excess samples from model.sample() to be wasted.

For this issue, I think the best solution is to refactor the relevant code so that this `batch_size` variable becomes a transferable context information, which can be perceived by both the model and the Synthesizer. Perhaps we could add a TODO.

Reply:
Yes. But it is not all model are sampling in batch.

How about the Change Description 4?
(Created a subclass for batched model)

Reply:
> Yes. But it is not all model are sampling in batch.
> 
> How about the Change Description 4? (Created a subclass for batched model)

I think it's good for now.